packages/angular/src/schematics/convert-tslint-to-eslint/convert.ts

```diff
@@ -0,0 +1,840 @@
+    templateRules,
+  };
+}
+
+function dedupeEnvAgainstConfigs(
+  convertedConfig: ESLintLinter.Config,
+  otherConfigs: ESLintLinter.Config[]
+) {
+  otherConfigs.forEach((againstConfig) => {
+    updateObjPropAndRemoveDuplication(
+      convertedConfig,
+      againstConfig,
+      'env',
+      true
+    );
+  });
+}
+
+function dedupeRulesAgainstConfigs(
+  convertedConfig: ESLintLinter.Config,
+  otherConfigs: ESLintLinter.Config[]
+) {
+  otherConfigs.forEach((againstConfig) => {
+    updateObjPropAndRemoveDuplication(
+      convertedConfig,
+      againstConfig,
+      'rules',
+      false
+    );
+  });
+}
+
+function dedupePluginsAgainstConfigs(
+  convertedConfig: ESLintLinter.Config,
+  otherConfigs: ESLintLinter.Config[]
+) {
+  otherConfigs.forEach((againstConfig) => {
+    updateArrPropAndRemoveDuplication(
+      convertedConfig,
+      againstConfig,
+      'plugins',
+      true
+    );
+  });
+}
+
+/**
+ * We don't want the user to depend on the TSLint fallback plugin, we will instead
+ * explicitly inform them of the rules that could not be converted automatically and
+ * advise them on what to do next.
+ */
+function warnInCaseOfUnconvertedRules(
+  context: SchematicContext,
+  tslintConfigPath: string,
+  unconvertedTSLintRules: TSLintRuleOptions[]
+): void {
+  /*
+   * The following rules are known to be missing from the Angular CLI equivalent TSLint
+   * setup, so they will be part of our convertedRoot data:
+   *
+   * // FORMATTING! Please use prettier y'all!
+   * "import-spacing": true
+   *
+   * // POSSIBLY NOT REQUIRED - typescript-eslint provides explicit-function-return-type (not yet enabled)
+   * "typedef": [
+   *    true,
+   *    "call-signature",
+   *  ]
+   *
+   * // FORMATTING! Please use prettier y'all!
+   *  "whitespace": [
+   *    true,
+   *    "check-branch",
+   *    "check-decl",
+   *    "check-operator",
+   *    "check-separator",
+   *    "check-type",
+   *    "check-typecast",
+   *  ]
+   */
+  const unconvertedTSLintRuleNames = unconvertedTSLintRules
+    .filter(
+      (unconverted) =>
+        !['import-spacing', 'whitespace', 'typedef'].includes(
+          unconverted.ruleName
+        )
+    )
+    .map((unconverted) => unconverted.ruleName);
+
+  if (unconvertedTSLintRuleNames.length > 0) {
+    context.logger.warn(
+      `\nWARNING: Within "${tslintConfigPath}", the following ${unconvertedTSLintRuleNames.length} rule(s) did not have known converters in https://github.com/typescript-eslint/tslint-to-eslint-config`
+    );
+    context.logger.warn('\n  - ' + unconvertedTSLintRuleNames.join('\n  - '));
+    context.logger.warn(
+      '\nYou will need to decide on how to handle the above manually, but everything else has been handled for you automatically.\n'
+    );
+  }
+}
+
+function likelyContainsTSLintComment(fileContent: string): boolean {
+  return fileContent.includes('tslint:');
+}
+
+function convertTSLintDisableCommentsForProject(projectName: string): Rule {
```

Comment:
You could use this method which will also not visit ignored directories and files:

https://github.com/nrwl/nx/blob/master/packages/workspace/src/utils/rules/visit-not-ignored-files.ts#L10

Reply:
Cool thanks

Reply:
I've added some previously missing test coverage for this logic too